core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -58,25 +61,33 @@ public AtomicFileWriter(File f) throws IOException {
     }
 
     /**
-     * @param encoding
-     *      File encoding to write. If null, platform default encoding is chosen.
+     * @param encoding File encoding to write. If null, platform default encoding is chosen.
+     *
+     * @deprecated Use {@link #AtomicFileWriter(Path, Charset)}
      */
-    public AtomicFileWriter(File f, String encoding) throws IOException {
-        File dir = f.getParentFile();
+    @Deprecated
+    public AtomicFileWriter(@Nonnull File f, String encoding) throws IOException {
+        this(f.toPath(), encoding == null ? Charset.defaultCharset() : Charset.forName(encoding));
```

Comment:
`f.toPath()`  can throw `InvalidPathException` which could cause calling threads to die as it is a newly unexpected exception.

I would put all of this into an init so that you can catch the exception and wrap it in an `IOException` so that you do not potentially cause threads to die due to this change.

Reply:
Done. Finally just created a wrapper for the `File.toPath()` method to throw `IOException` instead if an `InvalidPathException` is thrown on this call.